### PR TITLE
Remove mklml pre-install in ci

### PR DESCRIPTION
**Pull request by xinyu-intel**
## Description ##
This PR is to remove mklml pre-install in CI and use a downloaded one while building.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

@pengzhao-intel @TaoLv  @marcoabreu

### CODE REVIEW

**Comment by roywei** [comment]
@mxnet-label-bot add[CI, pr-awaiting-review]

File: ci/docker/runtime_functions.sh [hunk 1]
```diff
@@ -502,7 +502,6 @@ build_ubuntu_cpu_mkldnn_mkl() {
     make  \
         DEV=1                         \
         ENABLE_TESTCOVERAGE=1         \
-        USE_CPP_PACKAGE=1             \
```
Comment (marcoabreu) [inline ci/docker/runtime_functions.sh]: Could you elaborate?

File: ci/jenkins/Jenkins_steps.groovy [hunk 2]
```diff
@@ -34,7 +34,7 @@ mx_dist_lib = 'lib/libmxnet.so, lib/libmxnet.a, 3rdparty/dmlc-core/libdmlc.a, 3r
 mx_cmake_lib = 'build/libmxnet.so, build/libmxnet.a, build/3rdparty/dmlc-core/libdmlc.a, build/tests/mxnet_unit_tests, build/3rdparty/openmp/runtime/src/libomp.so'
 // mxnet cmake libraries, in cmake builds we do not produce a libnvvm static library by default.
 mx_cmake_lib_debug = 'build/libmxnet.so, build/libmxnet.a, build/3rdparty/dmlc-core/libdmlc.a, build/tests/mxnet_unit_tests'
-mx_cmake_mkldnn_lib = 'build/libmxnet.so, build/libmxnet.a, build/3rdparty/dmlc-core/libdmlc.a, build/tests/mxnet_unit_tests, build/3rdparty/openmp/runtime/src/libomp.so, build/3rdparty/mkldnn/src/libmkldnn.so.0'
+mx_cmake_mkldnn_lib = 'build/libmxnet.so, build/libmxnet.a, build/3rdparty/dmlc-core/libdmlc.a, build/tests/mxnet_unit_tests, build/3rdparty/openmp/runtime/src/libomp.so, build/3rdparty/mkldnn/src/libmkldnn.so.0, build/libmklml_intel.so'
```
Comment (xinyu-intel) [inline ci/jenkins/Jenkins_steps.groovy]: @marcoabreu I think `libmklml_intel.so` should be cached in this docker env. So I add it to here but it still can not be found while testing. Any suggestions for this?

File: Makefile [hunk 3]
```diff
@@ -69,14 +69,23 @@ ifeq ($(USE_MKLDNN), 1)
 	MKLDNNROOT = $(ROOTDIR)/3rdparty/mkldnn/build/install
 	MKLROOT = $(ROOTDIR)/3rdparty/mkldnn/build/install
 	export USE_MKLML = 1
+	CFLAGS += -DMXNET_USE_MKLDNN=1
```
Comment (TaoLv) [inline Makefile]: This is the first appearance of  `CFLAGS`.

File: Makefile [hunk 4]
```diff
@@ -69,14 +69,23 @@ ifeq ($(USE_MKLDNN), 1)
...
+	CFLAGS += -DUSE_MKL=1
+	CFLAGS += -I$(ROOTDIR)/src/operator/nn/mkldnn/
+	ifneq ($(MKLDNNROOT), $(MKLROOT))
```
Comment (TaoLv) [inline Makefile]: always true.

File: cmake/DownloadMKLML.cmake [hunk 5]
```diff
@@ -41,6 +41,9 @@ if(MSVC)
   message(STATUS "Setting MKLROOT path to ${MKLROOT}")
 
   include_directories(${MKLROOT}/include)
+  file(COPY ${MKLROOT}/lib/libiomp5md.dll DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
```
Comment (TaoLv) [inline cmake/DownloadMKLML.cmake]: @lebeg Could you help to review these changes?

File: cmake/DownloadMKLML.cmake [hunk 6]
```diff
@@ -68,6 +73,9 @@ elseif(UNIX)
   set(MKLROOT "${CMAKE_CURRENT_BINARY_DIR}/mklml/${MKL_NAME}")
   message(STATUS "Setting MKLROOT path to ${MKLROOT}")
   include_directories(${MKLROOT}/include)
+  file(COPY ${MKLROOT}/lib/libiomp5.so DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
+  file(COPY ${MKLROOT}/lib/libmklml_gnu.so DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
```
Comment (lebeg) [inline cmake/DownloadMKLML.cmake]: You need either of this libraries, but not both.

File: Makefile [hunk 7]
```diff
@@ -123,18 +128,6 @@ ifndef LINT_LANG
 	LINT_LANG="all"
 endif
 
-ifeq ($(USE_MKLDNN), 1)
-	CFLAGS += -DMXNET_USE_MKLDNN=1
-	CFLAGS += -DUSE_MKL=1
-	CFLAGS += -I$(ROOTDIR)/src/operator/nn/mkldnn/
-	ifneq ($(MKLDNNROOT), $(MKLROOT))
```
Comment (TaoLv) [inline Makefile]: Previously we have this check because `MKLROOT` might be changed when build mshadow.

File: Makefile [hunk 8]
```diff
@@ -69,14 +69,19 @@ ifeq ($(USE_MKLDNN), 1)
...
+	LDFLAGS += -L$(MKLDNNROOT)/lib -lmkldnn -lmklml_intel -Wl,-rpath,'$${ORIGIN}'
 endif
 
 include $(TPARTYDIR)/mshadow/make/mshadow.mk
 include $(DMLC_CORE)/make/dmlc.mk
 
 # all tge possible warning tread
 WARNFLAGS= -Wall -Wsign-compare
-CFLAGS = -DMSHADOW_FORCE_STREAM $(WARNFLAGS)
+CFLAGS += -DMSHADOW_FORCE_STREAM $(WARNFLAGS)
```
Comment (TaoLv) [inline Makefile]: if `USE_MKLDNN=0`, here will become the first appearance of `CFLAGS`.

**Comment by Roshrini** [comment]
@xinyu-intel Can you please address comments? Thanks!

**Comment by anirudhacharya** [comment]
@mxnet-label-bot update [pr-awaiting-response]

**Comment by sandeep-krishnamurthy** [comment]
@xinyu-intel - Thanks for your contributions. What is the next step in this PR?

**Comment by vandanavk** [comment]
@mxnet-label-bot add [CI]

**Comment by ankkhedia** [comment]
@xinyu-intel Could you please look into the review comments?

**Comment by anirudhacharya** [comment]
@xinyu-intel @lebeg @TaoLv ping for update and another round of reviews.

**Comment by karan6181** [comment]
@xinyu-intel  Thank you for your contribution! Could you please address the review comments? Thanks

**Pull request closed without merging**